### PR TITLE
Add documentation for some Iterator methods

### PR DESCRIPTION
**Pull request by davydovanton**

### CODE REVIEW

File: src/iterator.cr [hunk 1]
```diff
@@ -99,18 +99,33 @@ module Iterator(T)
...
+  # given block returns a true value.
   def select(&func : T -> U)
     Select(typeof(self), T, U).new(self, func)
   end
 
+  # Returns an array for all elements of enum for which the given
+  # block returns false.
   def reject(&func : T -> U)
     Reject(typeof(self), T, U).new(self, func)
   end
```
Comment (jhass) [inline src/iterator.cr]: "Returns an array" doesn't seem correct for these, they return new iterators :)

**Comment by jhass** [comment]
I think in general it would be good to highlight that they return new iterators, in the examples too (which are just not right in many cases, sorry :/)

**Comment by davydovanton** [comment]
@jhass okay, it's no problem :)
Bun could you show not right examples?

File: src/iterator.cr [hunk 2]
```diff
@@ -99,18 +99,33 @@ module Iterator(T)
     self
   end
 
+  # Returns a new array with the results of running block once for
+  # every element in enum.
+  #
+  # ```
+  # [1, 2, 3].map &.*(2) # => [2, 4, 6]
+  # ```
```
Comment (jhass) [inline src/iterator.cr]: This is calling `Array#map`, not `Iterator#map`, so if we actually call `Iterator#map` we get:

```cr
[1, 2, 3].each.map &.*(2) # => Iterator(T)::Map(Array(T)::ItemIterator(Int32), Int32, Int32)(@iter=#<Array(T)::ItemIterator(Int32):0x8652fc0 @array=[1, 2, 3], @index=0>, @func=#<(Int32 -> Int32):0x804b1a0>)
```

A better example might be

```cr
doubled = [1, 2, 3].each.map &.*(2)
doubled.next # => 2
doubled.next # => 4
doubled.next # => 6
```

or even just
```cr
[1, 2, 3].map(&.*(2)).to_a # => [2, 4, 6]
```

The basic oversight is true for all other examples.

Comment (asterite) [inline src/iterator.cr]: Yes, adding a `to_a` call should be enough, but I agree the docs should say "Returns an iterator that ..."

**Comment by PragTob** [comment]
@davydovanton are you looking to pick this up again? Otherwise I might take it and fix it :)

**Comment by davydovanton** [comment]
@PragTob yep, sorry, I have some problems with other project (deadline :sweat:). I'll fix this tomorrow

**Comment by PragTob** [comment]
@davydovanton no worries, just checking in :)

**Comment by jhass** [comment]
Ping, could free any time by now? :)

**Comment by davydovanton** [comment]
@jhass I'm so sorry. I completed my GSoC project. :disappointed: I'll update this in the coming days :sweat:

**Comment by PragTob** [comment]
@davydovanton if you don't make it in the coming days, any objections to me picking up this PR, rebasing it and adding other documentation for new methods? :) Want to get those docs in!

**Comment by asterite** [comment]
@davydovanton There's really no pressure in you doing this. If other things have more priority you should definitely do those! There are plenty of other things where you can later contribute.

@PragTob You can go ahead and submit a separate pull request for the iterator docs :-)

**Comment by davydovanton** [comment]
closed by #1352

**Pull request closed without merging**